lib/stm32/l4/rcc.c

```diff
@@ -326,7 +326,7 @@ void rcc_set_hpre(uint32_t hpre)
 void rcc_set_main_pll(uint32_t source, uint32_t pllm, uint32_t plln, uint32_t pllp,
 	uint32_t pllq, uint32_t pllr)
 {
-	RCC_PLLCFGR = (pllm << RCC_PLLCFGR_PLLM_SHIFT) |
+	RCC_PLLCFGR = (RCC_PLLCFGR_PLLM(pllm) << RCC_PLLCFGR_PLLM_SHIFT) |
```

Comment:
I don't like this one, it makes it different in parameters you pass compared to f2/f4/f7

Reply:
are you referring to source?
for f2/f4/f7 you have two different functions `rcc_set_main_pll_hsi()` and `rcc_set_main_pll_hse()`
for l4 there is MSI, HSI16 and HSE so we would have to make three equivalent functions...

or to RCC_PLLCFGR_PLLM(pllm)? my reasoning for this is that when you look at the clock tree and you decide for a M value it's not clear that the register expects the value - 1.

Reply:
I'm talking about the RCC_PLLCFGR_PLLM() macro indeed.  I see now looking at the f4 reference manual again what you're doing, and yeah, this is fine.  This can just come in by itself I think.

Reply:
I see thsi commit is already standalone. I'll pull it into my staging.